tests/documentation/DocGrammarBearTest.py

```diff
@@ -0,0 +1,167 @@
+                DocGrammarBear(section, Queue()),
+                'dummy_filename',
+                test_data,
+                **arguments) as results:
+
+            diff = Diff(test_data)
+            for result in results:
+                # Only the given test file should contain a patch.
+                self.assertEqual(len(result.diffs), 1)
+
+                diff += result.diffs['dummy_filename']
+
+        self.assertEqual(expected_data, diff.modified)
+
+    return test_function
+
+
+@generate_skip_decorator(DocGrammarBear)
+class DocGrammarBearTest(unittest.TestCase):
+
+    def test_check_prerequisites(self):
+        _shutil_which = shutil.which
+        try:
+            shutil.which = lambda *args, **kwargs: None
+            self.assertEqual(DocGrammarBear.check_prerequisites(),
+                             'java is not installed.')
+
+            shutil.which = lambda *args, **kwargs: 'path/to/java'
+            self.assertTrue(DocGrammarBear.check_prerequisites())
+        finally:
+            shutil.which = _shutil_which
+
+    test_spelling = test(
+        make_docstring(main_desc='Thiss is main descrpton.\n'),
+        make_docstring(main_desc='This is main description.\n'))
+
+    test_capitalize_sentence_start = test(
+        make_docstring(main_desc='this sentence starts with small letter\n'),
+        make_docstring(main_desc='This sentence starts with small letter\n'))
+
+    test_extra_whitespace = test(
+        make_docstring(main_desc='This sentence    has extra  white spaces\n'),
+        make_docstring(main_desc='This sentence has extra white spaces\n'))
+
+    test_apostrophe_comma = test(
+        make_docstring(main_desc='This sentence doesnt have an apostrophe\n'),
+        make_docstring(main_desc='This sentence doesn\'t have an '
+                                 'apostrophe\n'))
+
+    correct_docstring = make_docstring(
+        main_desc='This documentation has correct grammar.\n',
+        param_desc='Dummy description.\n',
+        return_desc='Return Nothing.\n')
+
+    test_correct_grammar = test(correct_docstring, correct_docstring)
+
+    test_disable_setting_UPPERCASE_SENTENCE_START = test(
+        make_docstring(main_desc='sentence starting with lowercase.\n',
+                       param_desc='dummy description.\n',
+                       return_desc='Nothing.\n'),
+        make_docstring(main_desc='sentence starting with lowercase.\n',
+                       param_desc='dummy description.\n',
+                       return_desc='Nothing.\n'),
+        {'languagetool_disable_rules': 'UPPERCASE_SENTENCE_START'})
+
+    # FRENCH_WHITESPACE adds a unicode space if it finds empty strings.
+    # which was breaking this test case.
+    test_language_french = test(
+        make_docstring(main_desc='il monte en haut si il veut.\n'),
+        make_docstring(main_desc='Il monte s’il veut.\n'),
+        {'locale': 'fr',
+         'languagetool_disable_rules': 'FRENCH_WHITESPACE'})
+
```

Comment:
This has been taken from
https://github.com/myint/language-check/blob/14bae49f6357c2bc592a11972c457665f7b30c28/test.py#L60